app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# :nodoc:
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :sign_post, expect: %i[new index]
+
+  def new
+    @post = Post.new
+  end
+
+  def index
+    @posts = Post.all
+  end
+
+  def show; end
+
+  def create
+    @post = current_user.post.build(permit_post)
+    if @post.save
+      flash[:success] = 'The Post you created was a Success!'
+      redirect_to posts_path(@post)
```

Comment:
this redirect is making your route look like `http://localhost:3000/posts.6` the `.6` is because you are passing it argument but you don't need to to get all posts just redirecting to `posts_path` is fine.